rotkehlchen/exchanges/binance.py

```diff
@@ -522,30 +526,108 @@ def _deserialize_asset_movement(self, raw_data: Dict[str, Any]) -> Optional[Asse
 
         return None
 
+    def _query_api_query_dict_within_time_delta(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+            time_delta_ts: Timestamp,
+            method: str,
+    ) -> List[Dict[str, Any]]:
+        """Request via `api_query_dict()` from `start_ts` `end_ts` using a time
+        delta (offset) less than `time_delta_ts`.
+
+        Be aware of:
+          - If `start_ts` equals zero, the Binance launch timestamp is used
+          (from BINANCE_LAUNCH_TS). This value is not stored in the
+          `used_query_ranges` table, but 0.
+          - Timestamps are converted to milliseconds.
+          - This function does not currently support limits.
+        """
+        if method == 'depositHistory.html':
+            query_schema = 'depositList'
+        elif method == 'withdrawHistory.html':
+            query_schema = 'withdrawList'
+        else:
+            raise AssertionError(f'Unexpected binance method case: {method}.')
+
+        results: List[Dict[str, Any]] = []
+        # Create required time references in milliseconds
+        time_delta_ts_ms = time_delta_ts * 1000 - 1  # less than in ms
+        end_ts_ms = end_ts * 1000
+        if start_ts == Timestamp(0):
+            start_ts_ms = Timestamp(BINANCE_LAUNCH_TS) * 1000
+        else:
+            start_ts_ms = start_ts * 1000
+
+        from_ts_ms = start_ts_ms
+        to_ts_ms = (
+            start_ts_ms + time_delta_ts_ms  # Case request with time delta
+            if end_ts_ms - start_ts_ms > time_delta_ts_ms
+            else end_ts_ms  # Case request without time delta (1 request)
+        )
+        while True:
+            options = {
+                'timestamp': ts_now_in_ms(),
+                'startTime': from_ts_ms,
+                'endTime': to_ts_ms,
+            }
+            result = self.api_query_dict(method, options=options)
+            results.extend(result.get(query_schema, []))
+            # Case stop requesting
+            if to_ts_ms == end_ts_ms:
+                break
+            # Case 1 more request: update `to_ts_ms` to `end_ts_ms`
+            elif to_ts_ms + time_delta_ts_ms > end_ts_ms:
```

Comment:
Sometimes it seems to me you like to complicate things that can be a lot simpler some times. This whole `elif` is not needed.

The entire thing can be as simple as:

```python
if to_ts_ms >= end_ts_ms:
    break

from_ts_ms = to_ts_ms + 1
to_ts_ms = to_ts_ms + time_delta_ts_ms
```

no need for complicated if/elif/else

Reply:
I disagree here, let me explain my point of view.

Let's simplify the example with just "days". The external API requires less than 90, and I'm calling our endpoint with:
- start_ts: 0
- end_ts: 200

My implementation will do the following calls:
**Call 1**
- start_ts: 0
- end_ts: 89
**Call 2**
- start_ts: 90
- end_ts: 179
**Call 3**
- start_ts: 180
- end_ts: **200**

Your implementation will do the following calls:
**Call 1**
- start_ts: 0
- end_ts: 89
**Call 2**
- start_ts: 90
- end_ts: 179
**Call 3**
- start_ts: 180
- end_ts: **269**

This will have multiple consequences:
- Misleading time references and logs. Endpoint `end_ts` param won't match external API call "end_ts".
- Potentially requesting more results than expected, and future timestamps (around 90 days in the future).

I could merge the `elif` with the `else` as both share the `from_ts_ms` logic, but I still prefer have two cases and be accurate with the `to_ts_ms`.

What do you think!?

Reply:
Please keep it simple.

But indeed I wrote the code as pseudocode at 02:00 at night. The example you give is an edge case it was no handling. It should be:

```python
if to_ts_ms >= end_ts_ms:
    break

from_ts_ms = to_ts_ms + 1
to_ts_ms = mint(to_ts_ms + time_delta_ts_ms, end_ts_ms)
```

Notice the change in the last line, to make sure to limit the `to_ts` to the end_ts.

> Misleading time references and logs. Endpoint end_ts param won't match external API call "end_ts".

As for this just place the log query appropriately.